flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/StreamTask.java

```diff
@@ -1451,6 +1454,7 @@ private void checkpointStreamOperator(StreamOperator<?> op) throws Exception {
 			.setTaskName(taskName)
 			.build(bufferWriter);
 		output.setMetricGroup(environment.getMetricGroup().getIOMetricGroup());
+		output.isUnalignedCheckpoint(checkpointingMode == CheckpointingMode.UNALIGNED);
```

Comment:
nit: for the PoC but I guess this could be moved to the `RecordWriter`'s constructor (maybe via the builder)?

Reply:
Exactly, the elegant way should be in the constructor via builder. For the PoC I take the shortcut way to avoid touching more tests changes.